### ISSUE TITLE
Tooltip: abilty to control wrapper z-index

### ISSUE THREAD

**Issue by daiagi** [oruga-ui/oruga#521]
### Description

at [KodaDot](https://kodadot.xyz) we have a situation where the entire body has a z-index: 100
this causes the tooltip to not be above the content
it would be very useful if the tooltip wrapper z-index could be controlled by scss/css var or via prop

![image](https://user-images.githubusercontent.com/22791238/236172695-4f3b0805-02ea-421d-a6c6-d2557b9a7fd9.png)

overriding the default using ```$tooltip-content-zindex``` doesn't produce the desired result

![image](https://user-images.githubusercontent.com/22791238/236173200-5d09d3b1-8203-4949-b55e-b2331134233d.png)

### Why Oruga need this feature

support edge cases where z-index:99 might not be suitable

**Comment by daiagi** [comment]
another nice to have feature would be font-size prop

**Comment by jtommy** [comment]
> another nice to have feature would be font-size prop

You can do it adding a class or overriding the default.



> ### Description
> at [KodaDot](https://kodadot.xyz) we have a situation where the entire body has a z-index: 100 this causes the tooltip to not be above the content it would be very useful if the tooltip wrapper z-index could be controlled by scss/css var or via prop
> 
> ![image](https://user-images.githubusercontent.com/22791238/236172695-4f3b0805-02ea-421d-a6c6-d2557b9a7fd9.png)
> 
> overriding the default using `$tooltip-content-zindex` doesn't produce the desired result
> 
> `![image](https://user-images.githubusercontent.com/22791238/236173200-5d09d3b1-8203-4949-b55e-b2331134233d.png)`
> 
> ### Why Oruga need this feature
> support edge cases where z-index:99 might not be suitable

Are you using append-to-body?

**Comment by daiagi** [comment]
Over riding the default is not good enough.
I would like to be able to have different font sizes at different places and mobile / desktop


I tried both with append to body and without.
But Tbh I didn't give that prop much thought

Should I be using it?

**Comment by daiagi** [comment]
- font size
Overriding the default is not good enough as we wrap oruga ui in our own component and I would like to be able to use different sizes in different scenarios

- z index

Tbh I haven't given append to body prop much thought
Should I be using it?

**Comment by jtommy** [comment]
Oruga set font size using css so you can edit it only using css, applying your class for example.

Z-index is set only using `append-to-body` , try to avoid to use it (Anyway it needs to be refactored i the core code)

**Comment by daiagi** [comment]
ok.
both issued are resolved for me if ```append-to-body="false"```

would be nice to have them for ```append-to-body="true"``` as well

**Comment by mlmoravek** [comment]
@jtommy 
I found out that the `updateAppendToBody()` method when using `append-to-body="true"` is only called once the `isActive` is set to true. However, when the `isActive` is set to false, the method is *not* called, so the z-index of the wrapper does not get back to -1. 

Can be found in isActive watcher, the if is only called when `isActive` changes to true:
```
 if (value && this.appendToBody) {
       this.updateAppendToBody()
}
```

It would be also create to add a custom class property for the wrapper when `append-to-body` is used, like a `appendBodyClass` or something like that, to configure custom classed for the wrapper as well.

**Comment by mlmoravek** [comment]
I'm going to remove the z-index from the inline style and add it to the CSS theme instead. That way it's easier to override.

**Comment by mlmoravek** [comment]
> I'm going to remove the z-index from the inline style and add it to the CSS theme instead. That way it's easier to override.

This is implemented in v0.8.0
